lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -213,14 +218,24 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     return decodeSourceLoc(OptLoc->getStartSourceLoc());
   }
 
+  SILLocation::DebugLoc sanitizeCodeViewDebugLoc(SILLocation::DebugLoc DLoc) {
+    if (Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView)
+      // When WinDbg finds two locations with the same line but different
+      // columns, the user must select an address when they break on that
+      // line. Also, clang does not emit column locations in CodeView for C++.
+      DLoc.Column = 0;
+    return DLoc;
+  }
+
   SILLocation::DebugLoc decodeDebugLoc(SILLocation Loc) {
     if (Loc.isDebugInfoLoc())
-      return Loc.getDebugInfoLoc();
+      return sanitizeCodeViewDebugLoc(Loc.getDebugInfoLoc());
```

Comment:
You probably want to sanitize both. FYI: the getDebugSourceLoc() case is what you get when you parse debug information serialized into textual `.sil` files.

Reply:
``decodeSourceLoc`` already calls ``sanitizeCodeViewDebugLoc`` so that would be redundant. I can add it if you think it would make the code more clear.